simpleclient_caffeine/src/main/java/io/prometheus/client/cache/caffeine/CacheMetricsCollector.java

```diff
@@ -137,27 +160,34 @@ public void clear(){
                 "Cache load duration: both success and failures", labelNames);
         mfs.add(cacheLoadSummary);
 
-        for(Map.Entry<String, Cache> c: children.entrySet()) {
-            List<String> cacheName = Arrays.asList(c.getKey());
+        for(final Map.Entry<String, Cache> c: children.entrySet()) {
+            
+            List<String> labelValues = new ArrayList<String>(customLabels.size() + 1) {{
+                add(c.getKey());
+                for (Map.Entry<String, String> entry : customLabels.entrySet()) {
+                    add(entry.getValue());
+                }
+            }};
```

Comment:
It may be a good idea to store label inside `children` value. This would require adding custom class as a value but since it is internal we can implement this without breaking changes.